chain/chain/src/store.rs

```diff
@@ -1853,6 +1880,19 @@ impl<'a> ChainStoreUpdate<'a> {
         self.trie_changes.push(trie_changes);
     }
 
+    pub fn remove_consolidated_state_changes(&mut self, block_hash: CryptoHash, shard_id: ShardId) {
+        self.remove_consolidated_state_changes.push((block_hash, shard_id));
+    }
+
+    pub fn add_consolidated_state_changes(
+        &mut self,
+        block_hash: CryptoHash,
+        shard_id: ShardId,
+        state_changes: ConsolidatedStateChanges,
+    ) {
+        self.add_consolidated_state_changes.insert((block_hash, shard_id), state_changes);
```

Comment:
This is an opportunity for an assertion (if, of course, it would be a valid one): 

```suggestion
        let prev = self.add_consolidated_state_changes.insert((block_hash, shard_id), state_changes);
        assert!(prev.is_none());
```

Reply:
Good point!